kornia/geometry/transform/thin_plate_spline.py

```diff
@@ -0,0 +1,148 @@
+from typing import Tuple
+
+import torch
+import torch.nn as nn
+
+from kornia.utils import _extract_device_dtype
+
+__all__ = [
+    "get_tps_parameters",
+    "warp_points_tps",
+    "warp_img_tensor_tps"
+]
+
+# utilities for computing thin plate spline transforms
+
+
+def _pair_square_euclidean(tensor1: torch.Tensor, tensor2: torch.Tensor) -> torch.Tensor:
+    """Compute the pairwise squared euclidean distance matrices (B, N, M) between two tensors
+        with shapes (B, N, C) and (B, M, C)."""
+    # ||t1-t2||^2 = (t1-t2)^T(t1-t2) = t1^T*t1 + t2^T*t2 - 2*t1^T*t2
+    t1_sq: torch.Tensor = tensor1.mul(tensor1).sum(dim=-1, keepdim=True)
+    t2_sq: torch.Tensor = tensor2.mul(tensor2).sum(dim=-1, keepdim=True).transpose(1, 2)
+    t1_t2: torch.Tensor = tensor1.matmul(tensor2.transpose(1, 2))
+    square_dist: torch.Tensor = -2 * t1_t2 + t1_sq + t2_sq
+    square_dist: torch.Tensor = square_dist.clamp(min=0)  # handle possible numerical errors
+    return square_dist
```

Comment:
what about using `torch.cdist` ?

https://pytorch.org/docs/stable/generated/torch.cdist.html?highlight=distance

Reply:
I didn't use `torch.cdist` for two reasons: 1) I found errors in the gradient when I was using it a while back (it may be fixed now, not sure), and 2) there's no option to return the sqaured distance instead of the standard distance. But if you think it would be better, I could use `cdist` and square it instead.

Reply:
ok, that's fine I guess. @ducha-aiki any comment on that ?